### PR TITLE
Update golang to 1.15.5

### PR DESCRIPTION
**Pull request by kuisathaverat**
This PR takes the changes in https://github.com/elastic/golang-crossbuild/pull/68 and add the changes in https://github.com/elastic/golang-crossbuild/pull/65 for go 1.14 and 1.15. Also, it runs all Go versions in parallel.

```
docker.elastic.co/beats-dev/golang-crossbuild      1.15.6-arm               fe1b24036a90   About an hour ago   1.32GB
docker.elastic.co/beats-dev/golang-crossbuild      1.15.6-arm-debian9       8af85b7fbb21   56 minutes ago      1.32GB
docker.elastic.co/beats-dev/golang-crossbuild      1.15.6-base              844fe6ea41c3   About an hour ago   735MB
docker.elastic.co/beats-dev/golang-crossbuild      1.15.6-base-debian10     2726ddaa0621   22 minutes ago      788MB
docker.elastic.co/beats-dev/golang-crossbuild      1.15.6-base-debian7      352ae7583f71   40 minutes ago      612MB
docker.elastic.co/beats-dev/golang-crossbuild      1.15.6-base-debian8      b190968ddb69   54 minutes ago      707MB
docker.elastic.co/beats-dev/golang-crossbuild      1.15.6-base-debian9      844fe6ea41c3   About an hour ago   735MB
docker.elastic.co/beats-dev/golang-crossbuild      1.15.6-darwin            57393537ba33   About an hour ago   1.31GB
docker.elastic.co/beats-dev/golang-crossbuild      1.15.6-darwin-debian10   07283ef0d88a   13 minutes ago      1.38GB
docker.elastic.co/beats-dev/golang-crossbuild      1.15.6-darwin-debian8    710c37b156c8   48 minutes ago      1.23GB
docker.elastic.co/beats-dev/golang-crossbuild      1.15.6-darwin-debian9    e18fd3b32b8d   56 minutes ago      1.31GB
docker.elastic.co/beats-dev/golang-crossbuild      1.15.6-main              829df18b0f5d   About an hour ago   1.72GB
docker.elastic.co/beats-dev/golang-crossbuild      1.15.6-main-debian10     a933adbb06ea   17 minutes ago      1.83GB
docker.elastic.co/beats-dev/golang-crossbuild      1.15.6-main-debian7      249fd5049fb6   38 minutes ago      1.04GB
docker.elastic.co/beats-dev/golang-crossbuild      1.15.6-main-debian8      22879d479bb7   53 minutes ago      1.42GB
docker.elastic.co/beats-dev/golang-crossbuild      1.15.6-main-debian9      a78eaac8b5a4   56 minutes ago      1.72GB
docker.elastic.co/beats-dev/golang-crossbuild      1.15.6-mips              ad5ad0b8691a   About an hour ago   1.44GB
docker.elastic.co/beats-dev/golang-crossbuild      1.15.6-mips-debian9      92202c1c4b8e   56 minutes ago      1.44GB
docker.elastic.co/beats-dev/golang-crossbuild      1.15.6-ppc               a71678d649d1   About an hour ago   1.26GB
docker.elastic.co/beats-dev/golang-crossbuild      1.15.6-ppc-debian9       df0a78b53e4a   56 minutes ago      1.26GB
docker.elastic.co/beats-dev/golang-crossbuild      1.15.6-s390x             16f6f2b38133   About an hour ago   1.08GB
docker.elastic.co/beats-dev/golang-crossbuild      1.15.6-s390x-debian9     003bf4ade12d   56 minutes ago      1.08GB
```

related to https://github.com/elastic/golang-crossbuild/issues/71

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,13 +1,14 @@
...
 	@$(foreach var,$(TARGETS), \
-		$(MAKE) -C $(var) $@ || echo '1' > ${status}; \
-		$(MAKE) -C $(var) -f Makefile.debian7 $@ || echo '1' > ${status}; \
-		$(MAKE) -C $(var) -f Makefile.debian8 $@ || echo '1' > ${status}; \
-		$(MAKE) -C $(var) -f Makefile.debian9 $@ || echo '1' > ${status};)
+		$(MAKE) -C $(var) $@; || echo '1' > ${status}; \
+		$(MAKE) -C $(var) -f Makefile.debian7 $@; || echo '1' > ${status}; \
+		$(MAKE) -C $(var) -f Makefile.debian8 $@; || echo '1' > ${status}; \
+		$(MAKE) -C $(var) -f Makefile.debian9 $@; || echo '1' > ${status}; \
+		$(MAKE) -C $(var) -f Makefile.debian10 $@; || echo '1' > ${status};) \
```
Comment (kuisathaverat) [inline Makefile:11]:
```suggestion
		$(MAKE) -C $(var) -f Makefile.debian10 $@; || echo '1' > ${status};)
```

File: Makefile [hunk 2]
```diff
@@ -19,7 +20,8 @@ push:
 		$(MAKE) -C $(var) $@ || echo '1' > ${status}; \
 		$(MAKE) -C $(var) -f Makefile.debian7 $@ || echo '1' > ${status}; \
 		$(MAKE) -C $(var) -f Makefile.debian8 $@ || echo '1' > ${status}; \
-		$(MAKE) -C $(var) -f Makefile.debian9 $@ || echo '1' > ${status};)
+		$(MAKE) -C $(var) -f Makefile.debian9 $@ || echo '1' > ${status}; \
+    $(MAKE) -C $(var) -f Makefile.debian10 $@ || echo '1' > ${status};)
```
Comment (kuisathaverat) [inline Makefile:24]:
```suggestion
		$(MAKE) -C $(var) -f Makefile.debian10 $@ || echo '1' > ${status};)
```

File: Jenkinsfile [hunk 3]
```diff
@@ -12,7 +12,7 @@ pipeline {
     DOCKER_REGISTRY_SECRET = 'secret/observability-team/ci/docker-registry/prod'
     REGISTRY = 'docker.elastic.co'
     STAGING_IMAGE = "${env.REGISTRY}/observability-ci"
-    GO_VERSION = '1.14.2'
+    GO_VERSION = '1.15.4'
```
Comment (urso) [inline Jenkinsfile:15]: 1.15.4 is already outdated. 1.15.5 is the current one.

**Comment by andresrc** [comment]
Whether we pink this or not in 7.11 under discussion: https://github.com/elastic/elastic-agent-client/issues/20

File: README.md [hunk 4]
```diff
@@ -14,16 +14,16 @@ The base image used is Debian 9 (stretch) unless otherwise specified.
...
-- `1.10.8-main`, `1.11.13-main`, `1.12.12-main`, `1.13.12-main`, `1.14.7` - linux/{amd64,386} and windows/{amd64,386}
-- `1.10.8-arm`, `1.11.13-arm`, `1.12.12-arm`, `1.13.12-arm`, `1.14.7` - linux/{armv5,armv6,armv7,arm64}
-- `1.10.8-darwin`, `1.11.13-darwin`, `1.12.12-darwin`, `1.13.12-darwin`, `1.14.7` - darwin/{amd64,386}
-- `1.10.8-ppc`, `1.11.13-ppc`, `1.12.12-ppc`, `1.13.12-ppc`, `1.14.7` - linux/{ppc64,ppc64le}
-- `1.10.8-mips`, `1.11.13-mips`, `1.12.12-mips`, `1.13.12-mips`, `1.14.7` - linux/{mips,mipsle,mips64,mips64le}
-- `1.10.8-s390x`, `1.11.13-s390x`, `1.12.12-s390`, `1.13.12-s390`, `1.14.7` - linux/s390x
-- `1.10.8-main-debian7`, `1.11.13-main-debian7`, `1.12.12-debian7`, `1.13.12-debian7`, `1.14.7` - linux/{amd64,386} and windows/{amd64,386} (Debian 7
+- `1.10.8-main`, `1.11.13-main`, `1.12.12-main`, `1.13.12-main`, `1.14.7-main`, `1.15.4-main` - linux/{amd64,386} and windows/{amd64,386}
+- `1.10.8-arm`, `1.11.13-arm`, `1.12.12-arm`, `1.13.12-arm`, `1.14.7-arm`, `1.15.4-arm` - linux/{armv5,armv6,armv7,arm64}
+- `1.10.8-darwin`, `1.11.13-darwin`, `1.12.12-darwin`, `1.13.12-darwin`, `1.14.7-darwin`, `1.15.3-darwin/amd64` - darwin/{amd64,386}
```
Comment (kvch) [inline README.md:19]: Nit: Golang 1.15 does not support `darwin/386` and `darwin/arm` anymore. Ref: https://golang.org/doc/go1.15#darwin

File: go1.14/arm/Dockerfile.tmpl [hunk 5]
```diff
@@ -26,30 +25,25 @@ RUN \
...
 COPY rootfs /
 
+# Basic test
+RUN cd / \
+  && aarch64-linux-gnu-gcc helloWorld.c -o helloWorld \
+  && file helloWorld \
+  && readelf -h helloWorld \
+  && file helloWorld | cut -d "," -f 2 | grep -c 'ARM aarch64'\
+  && rm helloWorld.c helloWorld
+
```
Comment (v1v) [inline go1.14/arm/Dockerfile.tmpl:46]: 🤟

File: Jenkinsfile [hunk 6]
```diff
@@ -35,37 +35,54 @@ pipeline {
...
+              branch 'master'
+            }
+            stages {
+              stage('Publish') {
+                steps {
+                  withGithubNotify(context: "Publish ${GO_FOLDER} ${MAKEFILE}") {
+                    publishImages()
+                  }
+                }
+              }
```
Comment (v1v) [inline Jenkinsfile:85]:
```suggestion
          stage('Release') {
            when {
              branch 'master'
            }
            steps {
              withGithubNotify(context: "Publish ${GO_FOLDER} ${MAKEFILE}") {
                publishImages()
              }
```

File: Jenkinsfile [hunk 7]
```diff
@@ -12,7 +12,7 @@ pipeline {
     DOCKER_REGISTRY_SECRET = 'secret/observability-team/ci/docker-registry/prod'
     REGISTRY = 'docker.elastic.co'
     STAGING_IMAGE = "${env.REGISTRY}/observability-ci"
-    GO_VERSION = '1.14.2'
+    GO_VERSION = '1.15.5'
```
Comment (urso) [inline Jenkinsfile:15]: Uhm... 1.15.6 (with a few critical fixes) was released right after you changed it to 1.15.5 :)

File: go1.14/main/Dockerfile.tmpl [hunk 8]
```diff
@@ -1,12 +1,12 @@
 ARG REPOSITORY
 ARG VERSION
-ARG TAG_EXTENSION=
+ARG TAG_EXTENSION
```
Comment (v1v) [inline go1.14/main/Dockerfile.tmpl:3]: Similar to the other templates

```suggestion
ARG TAG_EXTENSION=''
```

File: go1.15/main/Dockerfile.tmpl [hunk 9]
```diff
@@ -0,0 +1,42 @@
+ARG REPOSITORY
+ARG VERSION
+ARG TAG_EXTENSION
```
Comment (v1v) [inline go1.15/main/Dockerfile.tmpl:3]:
```suggestion
ARG TAG_EXTENSION=''
```

**Review by v1v — APPROVED** [review]
💯

**Review by cachedout — APPROVED** [review]

**Pull request merged**